luxonis_train/config/predefined_models/classification_model.py

```diff
@@ -14,19 +14,32 @@
 
 @dataclass
 class ClassificationModel(BasePredefinedModel):
-    backbone: str = "MicroNet"
+    variant: str | None = None
```

Comment:
Can we make the variants more similar to how we handle them in e.g. EfficientRep? So make predefined model an actual class (instead of dataclass) with all the parameters being None by default (except variant="lite" by default). Then have VariantLiteral type ["lite", "heavy"] and a function that gets specific parameters based on variant and uses them if others aren't specifid (e.g. uses "ResNet-18" as backbone if variant="lite" and no backbone is set, but if backbone is explicitly set then use that one.

The same can be done for all ther predefined models.